org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -190,6 +194,30 @@ public Charset getCharset() {
     public <G> G getClientObject() {
       return (G) file;
     }
+
+    @Override
+    public String contents() throws IOException {
+      try (TextFileContext context = new TextFileContext(file)) {
+        return context.getDocument().get();
+      } catch (CoreException e) {
+        // ignore
+      }
+      return null;
+    }
+
+    @Override
+    public InputStream inputStream() throws IOException {
+      Charset charset = getCharset();
+      if (charset == null) {
+        charset = StandardCharsets.UTF_8;
+      }
+      try (TextFileContext context = new TextFileContext(file)) {
+        return new ByteArrayInputStream(contents().getBytes(charset));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'contents' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)